### PR TITLE
Updates for OEP-55 (Maintainership) conformance

### PR DESCRIPTION
**Pull request by nedbat**
@feanil I have a number of questions about what I've done here....

### CODE REVIEW

**Comment by nedbat** [comment]
When I was creating my catalog-info.yaml file from the sample in the ADR, I had these questions:

```
# (Required) Acceptable Values: Component, Resource, System
kind: Component
```
The help for `kind` links to https://backstage.io/docs/features/software-catalog/system-model, which feels very abstract.  We could use more guidance on how to interpret it for this use.  I have a feeling that every repo will be "Component".

```
metadata:
  name: 'docs.openedx.org'
  description: "This repository contains the root documentation site for https://docs.openedx.org"
```

A nit: I thought the example would be better without "This repository contains", but then I wondered if the system model would mean some catalog items wouldn't be repos at all, and then I didn't know if it was important to indicate it was a repository in the description.

```
  links:
    - url: "https://docs.openedx.org"
      title: "Deployed Site"
      icon: "Web"
```

What links will a repo typically have?  This feels like one of many places in catalog-info.yaml that could duplicate metadata already present in setup.py.  For example, it might seem natural to have a link here to the published docs for a repo, but do we want to duplicate information like that?

```
  annotations:
    # (Optional) Annotation keys and values can be whatever you want.
    # We use it in Open edX repos to have a comma-separated list of GitHub user
    # names that might be interested in changes to the architecture of this
    # component.
    openedx.org/arch-interest-groups: "feanil"
```

I don't know who to name here.  I left `arch-interest-groups` as an empty string.

```
spec:
  # (Required) This can be a group(`group:<group_name>` or a user(`user:<github_username>`)
  owner: group:docs.edx.org-maintainers
```

We've touched on group owners vs people owners a number of times, and seem to have come to a middle ground: there is value in naming both a group and a contact person.  How would I do that in this file?

```
  # (Optional) The value can be the name of any known component.
  subcomponentOf: '<name_of_a_component>'

  # (Optional) An array of different components or resources.
  dependsOn:
  - '<component_or_resource>'
  - '<another_component_or_resource>'
```

I'm not sure what to put in these sections: again, `dependsOn` seems like it duplicates the `requirements/*` files, or maybe it's a different kind of dependency that I don't understand yet? Guidance would be appreciated.

**Comment by feanil** [comment]
> When I was creating my catalog-info.yaml file from the sample in the ADR, I had these questions:
> 
> ```
> # (Required) Acceptable Values: Component, Resource, System
> kind: Component
> ```
> 
> The help for `kind` links to https://backstage.io/docs/features/software-catalog/system-model, which feels very abstract. We could use more guidance on how to interpret it for this use. I have a feeling that every repo will be "Component".

I agree, I think all repos will be components, I haven't gotten familiar enough with the model to figure out what we need to do to have other `kind` types. But that can be general guidance that we can provide in the example.

> 
> ```
> metadata:
>   name: 'docs.openedx.org'
>   description: "This repository contains the root documentation site for https://docs.openedx.org"
> ```
> 
> A nit: I thought the example would be better without "This repository contains", but then I wondered if the system model would mean some catalog items wouldn't be repos at all, and then I didn't know if it was important to indicate it was a repository in the description.

Yes, I believe there will be items that are not repositories so it makes sense to mention that.
> 
> ```
>   links:
>     - url: "https://docs.openedx.org"
>       title: "Deployed Site"
>       icon: "Web"
> ```
> 
> What links will a repo typically have? This feels like one of many places in catalog-info.yaml that could duplicate metadata already present in setup.py. For example, it might seem natural to have a link here to the published docs for a repo, but do we want to duplicate information like that?

I don't think we have enough data to know what should be typical yet. It makes sense to be concerned that this would have some links that are also duplicated in the setup.py for python libraries but it may be worth it to do that duplication anyway since those links don't really change very often. It makes sense to want to link to the published docs and that will probably be true whether or not it's a python library.

> 
> ```
>   annotations:
>     # (Optional) Annotation keys and values can be whatever you want.
>     # We use it in Open edX repos to have a comma-separated list of GitHub user
>     # names that might be interested in changes to the architecture of this
>     # component.
>     openedx.org/arch-interest-groups: "feanil"
> ```
> 
> I don't know who to name here. I left `arch-interest-groups` as an empty string.
That's fine.  I think we'll have more guidance for it as the Arch Coordination group gets rolling.

> 
> ```
> spec:
>   # (Required) This can be a group(`group:<group_name>` or a user(`user:<github_username>`)
>   owner: group:docs.edx.org-maintainers
> ```
> 
> We've touched on group owners vs people owners a number of times, and seem to have come to a middle ground: there is value in naming both a group and a contact person. How would I do that in this file?

I believe owner only takes one value unfortunately, so I'm not sure we can easily indicate both, other than maybe through links for now?  eg. team as owner, and then a link to the main contact persion if that makes sense.

> 
> ```
>   # (Optional) The value can be the name of any known component.
>   subcomponentOf: '<name_of_a_component>'
> 
>   # (Optional) An array of different components or resources.
>   dependsOn:
>   - '<component_or_resource>'
>   - '<another_component_or_resource>'
> ```
> 
> I'm not sure what to put in these sections: again, `dependsOn` seems like it duplicates the `requirements/*` files, or maybe it's a different kind of dependency that I don't understand yet? Guidance would be appreciated.

I think this is useful for how this component is architecturally connected to other components, I don't think it has to be comprehensive but a more general, "if you're gonna use this, know that you'll need that".  It may be less relevant for libraries than it will be for services which have inter-dependencies.  Since those or not as obviously documented.

File: edx_rest_api_client/auth.py [hunk 1]
```diff
@@ -58,26 +58,38 @@ def __call__(self, r):
 
 
 class SuppliedJwtAuth(AuthBase):
-    """Attaches a supplied JWT to the given Request object."""
+    """
+    Attaches a supplied JWT to the given Request object.
+    """
```
Comment (robrap) [inline edx_rest_api_client/auth.py:63]: @nedbat: Are most of these changes just personal preference, or was this breaking a rule? Most of these seem fine according to https://peps.python.org/pep-0257/.

File: catalog-info.yaml [hunk 2]
```diff
@@ -0,0 +1,19 @@
...
+  description: "Utility library to simplify REST API access"
+  annotations:
+    # (Optional) Annotation keys and values can be whatever you want.
+    # We use it in Open edX repos to have a comma-separated list of GitHub user
+    # names that might be interested in changes to the architecture of this
+    # component.
+    openedx.org/arch-interest-groups: ""
+spec:
+  # (Required) This can be a group(`group:<group_name>` or a user(`user:<github_username>`)
+  owner: user:nedbat
```
Comment (robrap) [inline catalog-info.yaml:17]: Is this temporary? Or just part of your question on how to list an individual and a team? This is owned by arch-bom, right?

**Review by robrap — COMMENTED** [review]
Generally looks ok. I can't speak to correctness of the catalog file.

File: README.rst [hunk 3]
```diff
@@ -48,19 +49,74 @@ The value of the ``timeout`` setting is the same as for any request made with th
...
+    :target: https://codecov.io/github/edx/edx-rest-api-client?branch=master
+    :alt: Code coverage
+
+.. |status-badge| image:: https://img.shields.io/badge/Status-Maintained-brightgreen
+    :alt: Maintained
+
+.. |license-badge| image:: https://img.shields.io/github/license/openedx/edx-rest-api-client.svg
+    :target: https://github.com/openedx/edx-rest-api-client/blob/master/LICENSE
+    :alt: License
+.. |pypi-badge| image:: https://img.shields.io/pypi/v/edx-rest-api-client.svg
```
Comment (robrap) [inline README.rst:120]: Nit:
```suggestion

.. |pypi-badge| image:: https://img.shields.io/pypi/v/edx-rest-api-client.svg
```

**Comment by robrap** [comment]
In terms of duplication of a documentation link, might we want the README to simply link to backstage, with a note about what can be found there, and potentially drop duplicated links/metadata?

File: edx_rest_api_client/auth.py (hunk 1 above)
Comment (nedbat) [reply edx_rest_api_client/auth.py:63]: A number of the docstrings used single-line but with extra spaces at the end, and some were multi-line.  I made them all consistent.

File: catalog-info.yaml (hunk 2 above)
Comment (nedbat) [reply catalog-info.yaml]: I've changed the entry to `owner: group:openedx/arch-bom`, though I still have an open question about when we want to talk about teams and when we want to talk about individuals.

@feanil: note that groups should include the GitHub organization, and a few of our existing catalog-info.yaml files don't include the "group:" prefix:
```
openedx/openedx-events/catalog-info.yaml
13:  owner: group:hooks-extension-framework

openedx/DoneXBlock/catalog-info.yaml
14:  owner: tcril-engineering

openedx/openedx-backstage/catalog-info.yaml
12:  owner: backstage-maintainers

openedx/openedx-translations/catalog-info.yaml
14:  owner: tcril-engineering

openedx/openedx-learning/catalog-info.yaml
8:  owner: tcril-engineering
```

File: README.rst (hunk 3 above)
Comment (nedbat) [reply README.rst:121]: Thanks, I've added the blank line.

**Comment by nedbat** [comment]
> In terms of duplication of a documentation link, might we want the README to simply link to backstage, with a note about what can be found there, and potentially drop duplicated links/metadata?

This repo has no link to documentation anywhere yet, but a more filled-out repo would already have two links: one in the README, and one in the setup.py metadata for the PyPI page.  I don't think it will help users to funnel them through Backstage.  I think Feanil is right that the duplication is not a major concern, because the links rarely change.

Also, if we really wanted to, we could lint that the duplication is accurate.

**Comment by feanil** [comment]
> > > ```
> > > metadata:
> > >   name: 'docs.openedx.org'
> > >   description: "This repository contains the root documentation site for https://docs.openedx.org"
> > > ```
> > > 
> > > 
> > >     
> > >       
> > >     
> > > 
> > >       
> > >     
> > > 
> > >     
> > >   
> > > A nit: I thought the example would be better without "This repository contains", but then I wondered if the system model would mean some catalog items wouldn't be repos at all, and then I didn't know if it was important to indicate it was a repository in the description.
> > 
> > 
> > Yes, I believe there will be items that are not repositories so it makes sense to mention that.
> 
> Even so, I think we shouldn't use this construct as an example, just as function docstrings shouldn't start with "This function ....". I think this expresses it well:
> 
> ```
> description: "The source of the root documentation site...etc"
> ```
> 
> and even this does since it is obvious that the resource is a repo?
> 
> ```
> description: "The root documentation site...etc"
> ```

Fair enough, I think that's fine.  Once you click through it's pretty obvious what the thing is for now.  Do you want to update the sample text?

**Comment by nedbat** [comment]
> Fair enough, I think that's fine. Once you click through it's pretty obvious what the thing is for now. Do you want to update the sample text?

Sure: https://github.com/openedx/open-edx-proposals/pull/378

File: catalog-info.yaml (hunk 2 above)
Comment (nedbat) [reply catalog-info.yaml]: Further discussion (https://github.com/openedx/open-edx-proposals/pull/378) seems to indicate that the prefix is not needed, so I'm going with that for now.

**Pull request merged**